### ISSUE TITLE
New lint: enforce import renaming

### ISSUE THREAD

**Issue by DevinR528** [rust-lang/rust-clippy#7276]
### What it does

Check imports are renamed according to a config list (similar to [disallowed_method](https://rust-lang.github.io/rust-clippy/master/index.html#disallowed_method)).

### Categories

- Nursery

This is a style lint so consistency and readability (subjective).

### Drawbacks

None :shrug:

### Example

```rust
use serde_json::Value;
```

Could be written as:

```rust
use serde_json::Value as JsonValue;
```

If everyone is ok with this in the nursery I can open a PR.

**Comment by xFrednet** [comment]
This lint should fit into the groups:
* `restriction`: *for things which are usually not considered "bad", but may be useful to turn on in specific cases*
* or `pedantic`: *lints which are rather strict or might have false positives*

`disallowed_method` is only in nursery as it has some problems that need to be worked out.
